extensions/src/main/java/org/mapstruct/extensions/spring/converter/ConversionServiceAdapterGenerator.java

```diff
@@ -39,11 +40,29 @@ protected TypeSpec createMainTypeSpec(
     return adapterClassTypeSpec
         .addAnnotation(COMPONENT_ANNOTATION_CLASS_NAME)
         .addField(conversionServiceFieldSpec)
+            .addFields(buildTypeDescriptorFields(descriptor, conversionServiceFieldSpec))
         .addMethod(buildConstructorSpec(descriptor, conversionServiceFieldSpec))
         .addMethods(buildMappingMethods(descriptor, conversionServiceFieldSpec))
         .build();
   }
 
+  private List<FieldSpec> buildTypeDescriptorFields(ConversionServiceAdapterDescriptor descriptor, FieldSpec conversionServiceFieldSpec)
+  {
+      return descriptor.getFromToMappings().stream()
+              .flatMap(fromToMapping -> Stream.of(fromToMapping.getSource(), fromToMapping.getTarget()))
+              .distinct()
+              .map(typeName -> FieldSpec.builder(TYPE_DESCRIPTOR_CLASS_NAME, fieldName(typeName), PRIVATE, FINAL)
+                          .initializer(String.format("%s", typeDescriptorFormat(typeName)), typeDescriptorArguments(typeName).toArray()).build())
+              .collect(toList());
+  }
+
+  private String fieldName(TypeName typeName) {
+      return "typeDescriptor_" + typeName.toString()
```

Comment:
Can we please have the field names in SCREAMING_SNAKE_CASE? Aside from being all uppercase, ideally this would include dropping `_` characters at the end.